packages/@uppy/react/src/useDropzone.ts

```diff
@@ -0,0 +1,31 @@
+import { useContext, useMemo } from 'react'
+import type {
+  DragEvent as ReactDragEvent,
+  ChangeEvent as ReactChangeEvent,
+} from 'react'
+import {
+  createDropzone,
+  type DropzoneOptions,
+  type DropzoneReturn,
+} from '@uppy/components'
+import { UppyContext } from './headless/UppyContextProvider.js'
+
+type TDragEvent = DragEvent & ReactDragEvent<HTMLDivElement>
+type TChangeEvent = Event & ReactChangeEvent<HTMLInputElement>
+
+export function useDropzone(
+  options?: DropzoneOptions,
+): DropzoneReturn<TDragEvent, TChangeEvent> {
+  const ctx = useContext(UppyContext)
+
+  if (!ctx.uppy) {
+    throw new Error('useDropzone must be called within a UppyContextProvider')
+  }
+
+  const dropzone = useMemo(
+    () => createDropzone<TDragEvent, TChangeEvent>(ctx, options),
+    [ctx, options],
+  )
```

Comment:
`options` is an object that gets recreated and therefore changes on every render (because the caller calls `useDropZone({ noClick: true })`). This means that the useMemo hook gets also gets called on every render (because it has `options` in its dependencies, and because dependencies are only shallow compared). We should either:
1. remove the `useMemo`, because it is not necessary and confusing
2. or rewrite the code so that we compare each individual option value from the object (probably better performance than 1 because we don't need to call `createDropzone` on every render)

same goes for useFileInput.

```suggestion
  const dropzone = createDropzone<TDragEvent, TChangeEvent>(ctx, options);
```

```suggestion
  const { noClick, onDragOver, onDragEnter, onDragLeave, onDrop, onFileInputChange } = options;
  const dropzone = useMemo(
    () => createDropzone<TDragEvent, TChangeEvent>(ctx, { noClick, onDragOver, onDragEnter, onDragLeave, onDrop, onFileInputChange }),
    [ctx, noClick, onDragOver, onDragEnter, onDragLeave, onDrop, onFileInputChange],
  )
```